tests/link/pytorch/test_subtensor.py

```diff
@@ -0,0 +1,200 @@
+    assert isinstance(out_pt.owner.op, pt_subtensor.Subtensor)
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np])
+
+    a_pt = ps.int64("a")
+    a_np = 1
+    out_pt = x_pt[a_pt, 2, a_pt:2]
+
+    assert isinstance(out_pt.owner.op, pt_subtensor.Subtensor)
+    out_fg = FunctionGraph([x_pt, a_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np, a_np])
+
+    # Negative step
+    with pytest.raises(
+        NotImplementedError, match="Negative step sizes are not supported in Pytorch"
+    ):
+        out_pt = x_pt[::-1]
+        out_fg = FunctionGraph([x_pt], [out_pt])
+        compare_pytorch_and_py(out_fg, [x_np])
+
+
+def test_pytorch_AdvSubtensor():
+    shape = (3, 4, 5)
+    x_pt = pt.tensor("x", shape=shape, dtype="int")
+    x_np = np.arange(np.prod(shape)).reshape(shape)
+    # Advanced indexing
+    out_pt = pt_subtensor.advanced_subtensor1(x_pt, [1, 2])
+    assert isinstance(out_pt.owner.op, pt_subtensor.AdvancedSubtensor1)
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np])
+
+    out_pt = x_pt[[1, 2], [2, 3]]
+    assert isinstance(out_pt.owner.op, pt_subtensor.AdvancedSubtensor)
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np])
+
+    # Advanced and basic indexing
+    out_pt = x_pt[[1, 2], :]
+    assert isinstance(out_pt.owner.op, pt_subtensor.AdvancedSubtensor)
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np])
+
+    out_pt = x_pt[[1, 2], :, [3, 4]]
+    assert isinstance(out_pt.owner.op, pt_subtensor.AdvancedSubtensor)
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np])
+
+    out_pt = x_pt[[1, 2], None]
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np])
+
+    a_pt = ps.int64("a")
+    a_np = 2
+
+    out_pt = x_pt[[1, a_pt], a_pt]
+    out_fg = FunctionGraph([x_pt, a_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np, a_np])
+
+    out_pt = x_pt[np.random.binomial(1, 0.5, size=(3, 4, 5)).astype(bool)]
+    out_fg = FunctionGraph([x_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np])
+
+    a_pt = pt.tensor3("a", dtype="bool")
+    a_np = np.random.binomial(1, 0.5, size=(3, 4, 5)).astype(bool)
+    out_pt = x_pt[a_pt]
+    out_fg = FunctionGraph([x_pt, a_pt], [out_pt])
+    compare_pytorch_and_py(out_fg, [x_np, a_np])
+
+
+def test_pytorch_IncSubtensor():
+    rng = np.random.default_rng(42)
+
+    x_np = rng.uniform(-1, 1, size=(3, 4, 5)).astype(config.floatX)
+    x_pt = pt.constant(np.arange(3 * 4 * 5).reshape((3, 4, 5)).astype(config.floatX))
+
+    # "Set" basic indices
+    st_pt = pt.as_tensor_variable(np.array(-10.0, dtype=config.floatX))
+    out_pt = pt_subtensor.set_subtensor(x_pt[1, 2, 3], st_pt)
+    assert isinstance(out_pt.owner.op, pt_subtensor.IncSubtensor)
+    out_fg = FunctionGraph([], [out_pt])
+    compare_pytorch_and_py(out_fg, [])
```

Comment:
Don't test cases that are constant (don't need any `numpy` arguments in `compare_pytorch_and_py`), because we may not end up testing the implementation in pytorch at all